### DISCUSSION TITLE
Not able to read the Siemens WinCC SCADA server state

### DISCUSSION THREAD

**Discussion by rajnishnationfirst** [FreeOpcUa/python-opcua#1232] [Q&A]
Hi,

I am facing too much exceptions , like :

**[WinError 10053] An established connection was aborted by the software in your host machine**

I used to read the server state and accordingly take actions .

I am using the below code to get the server status :

Client URL:
```
hda_client = Client("opc.tcp://WINCCSERVERA:4862")



server=Server()
server.set_endpoint("opc.tcp://WINCCSERVERA:4862")
server_state_info = server.get_node("i=2256").get_data_value()
server_status = str(server_state_info.Value.Value.State)
print('Upper Blok',server_state_info, type(str(server_state_info)))
```

After i shut downt my WINCC SCADA , still i m getting the same status as **State:ServerState.Running**

How can i get my status correctly using free opcua.

**Main idea is , my program should keep continuous running and check server state before every request and if server state is STOP , then not send requests and if connection is available then make request.**

Pls help anyone .

Thanks and Regards

Rajnish Vishwakarma

**Comment by AndreasHeine** [comment]
what is this:
```
hda_client = Client("opc.tcp://WINCCSERVERA:4862")



server=Server()
server.set_endpoint("opc.tcp://WINCCSERVERA:4862")
server_state_info = server.get_node("i=2256").get_data_value()
server_status = str(server_state_info.Value.Value.State)
print('Upper Blok',server_state_info, type(str(server_state_info)))
```

are you server? are you client? whats the point... your code shows that your server-class-instance checks its own serverstate!?

> [WinError 10053] An established connection was aborted by the software in your host machine

this error does not come from python-opcua library and is more likely a networkproblem which you dont catch in your program!

**Comment by rajnishnationfirst** [comment]
Thanks Andreas, 

I am client.

How can I check my sever state bearing URL given in the below code. 

hda_client = Client("opc.tcp://WINCCSERVERA:4862")


It will be good of you if you can help me on how can I check my server state.. 

He me acting as a client. 

Thanks 

Rajnish Vishwakarma

**Comment by rajnishnationfirst** [comment]
Thanks Andrea's 

Here I m acting as a client with below the URL 

hda_client = Client("opc.tcp://WINCCSERVERA:4862")

How can read my server status having the URL of OPCUA server as 

"opc.tcp://WINCCSERVERA:4862"

It will be good of you if you can help me... On how can I read my server status as I have tired using the below code but getting RUNNING status for both when STOPPED AND RUNNING For the SERVR state node. 


server_state_info = server.get_node("i=2256").get_data_value()
server_status = str(server_state_info.Value.Value.State)
print('Upper Blok',server_state_info, type(str(server_state_info)))

Plz help with correct methods to use. 

Thanks 

Rajnish Vishwakarma

**Reply by AndreasHeine** [reply]
you need to read the node value cyclic to check the status if the read fails (network issue or server is not avalible) then you need to catch the error and overwrite the serverstate to not running on clientside maybe build some logic to atempt a reconnect after some time but this is an architectural thing in you app-design!

**Reply by zerox1212** [reply]
How do you expect a stopped server to respond to your client if it's stopped? Come on now... lol
